### PR TITLE
work on the control channel and adopting TLS 1.3 handshake and alert …

### PR DESCRIPTION
**Pull request by mmosko**
…message types.

### CODE REVIEW

File: draft-wood-icnrg-tlvencap.mkd [hunk 1]
```diff
@@ -77,6 +77,10 @@ It is a requirement of ESIC that one inner packet fit in one outer packet.  This
 because ESIC does not define a method to issue extra outer interests to fetch extra
 outer content objects.  It relies entirely on Interests generated by the consumer application.
 
+ESIC defines a control channel within the out context by using special names
```
Comment (chris-wood) [inline draft-wood-icnrg-tlvencap.mkd]: "within the outer context"

File: draft-wood-icnrg-tlvencap.mkd [hunk 2]
```diff
@@ -143,9 +152,14 @@ prefix as used in the key exchagne or it is derived from within CCNxKE from Pref
...
 
 ~~~
-OuterName := '/' PREFIX '/' SID=sid '/' CHUNK=chunk
+OuterName := ccnx:/PREFIX/SID=sid/CHUNK=chunk
 ~~~
 
+Chunk numbers are limited to 8 bytes and do not wrap around.  When the consumer gets near the
+end of the sequence number space, it must request a re-keying via the control channel.  Because CCNx in
+a pull-driven model, the consumer is reponsible for the chunk number and thus responsible for requesting
+the re-keying.
```
Comment (chris-wood) [inline draft-wood-icnrg-tlvencap.mkd]: We should consider cases where the producer might want to trigger a re-key

File: draft-wood-icnrg-tlvencap.mkd [hunk 3]
```diff
@@ -282,9 +296,111 @@ Protocol of the Deframer/Decrypter:
    - Pass decrypted packet to Sink/Source (or decompressor)
 
 
+# Control Channel
+
+The consumer and producer will need to exchange signaling about the encryption context.
+It should not be obvious to a 3rd party observing the outer packets what is control signaling
```
Comment (chris-wood) [inline draft-wood-icnrg-tlvencap.mkd]: "what is control and what is data..."

Comment (chris-wood) [inline draft-wood-icnrg-tlvencap.mkd]: Or maybe just say, "control and data traffic must be indistinguishable."

File: draft-wood-icnrg-tlvencap.mkd [hunk 4]
```diff
@@ -282,9 +296,111 @@ Protocol of the Deframer/Decrypter:
...
+outer name available.  The producer can only send the consumer a control message if the
+consumer has outstanding Interests up to the producer.  If there is one or more oustanding
+interests in the outer name space, the producer normally would send a Content Object or Interest Return
+corresponding to some inner name.  In this case, the producer would instead inject a control
+packet Interest in the downstream.  This means the producer is now short one outer Interest in the
+upstream direction.  Therefore, whenever the Deframer/Decrypter sees a control message
+in the downstream direction, it MUST insert an upstream "no-op" packet, padded out to
+statistically undetectable length, to give the producer back a missing name slot.
+
+We allow one one ESIC control packet in one outer packet.  However, we allow multiple
```
Comment (chris-wood) [inline draft-wood-icnrg-tlvencap.mkd]: "one one"

File: draft-wood-icnrg-tlvencap.mkd [hunk 5]
```diff
@@ -282,9 +296,111 @@ Protocol of the Deframer/Decrypter:
...
+Alert messages to be encoded in the payload.
```
Comment (chris-wood) [inline draft-wood-icnrg-tlvencap.mkd]: "...so long as it remains indistinguishable from packets in the given _direction_"

File: draft-wood-icnrg-tlvencap.mkd [hunk 6]
```diff
@@ -282,9 +296,111 @@ Protocol of the Deframer/Decrypter:
...
+
+TODO: Add examples with loss
+
+## ESIC Control Messages
+
+ESIC adopts the TLS 1.3 Alert Protocol for its control messages.  The TLV type of the message
+inside the control packet payload is taken from the enum AlertDescription.  As per TLS 1.3,
+fatal Alert messages are an immediate close of the ESIC session.
+
+As per TLS 1.3, each party MUST send a close_nofity message closeing the write side of
```
Comment (chris-wood) [inline draft-wood-icnrg-tlvencap.mkd]: "closeing" -> "closing"

File: draft-wood-icnrg-tlvencap.mkd (hunk 2 above)
Comment (mmosko) [inline draft-wood-icnrg-tlvencap.mkd]: The producer can always request a re-key, but it is the consumer that controls the chunk number space, so its up to the consumer to ensure that there's a rekey before running out of chunk numbers.

Comment (chris-wood) [inline draft-wood-icnrg-tlvencap.mkd]: Right -- the text seems to imply that only the consumer can request a re-key. I'd word it as:

"the consumer is responsible for the chunk number and thus responsible for requesting the re-keying in the event that the chunk number space is exhausted."

**Comment by mmosko** [comment]
Updated commit with fixes to comments.

**Pull request merged**